File: examples/v2/vcs-integrations/go/main.go
package main

import (
	integrations "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/integrations"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		githubID := cfg.Get("githubIntegrationId")
		if githubID == "" {
			githubID = "gh-org-integration"
		}
		githubEnterpriseID := cfg.Get("githubEnterpriseIntegrationId")
		if githubEnterpriseID == "" {
			githubEnterpriseID = "ghe-org-integration"
		}
		gitlabID := cfg.Get("gitlabIntegrationId")
		if gitlabID == "" {
			gitlabID = "gl-org-integration"
		}
		bitbucketID := cfg.Get("bitbucketIntegrationId")
		if bitbucketID == "" {
			bitbucketID = "bb-org-integration"
		}
		azureDevOpsID := cfg.Get("azureDevOpsIntegrationId")
		if azureDevOpsID == "" {
			azureDevOpsID = "ado-org-integration"
		}

		if _, err := integrations.NewGitHubIntegration(ctx, "github", &integrations.GitHubIntegrationArgs{
			OrgName:                     pulumi.String(serviceOrg),
			IntegrationId:               pulumi.String(githubID),
			DisablePRComments:           pulumi.Bool(false),
			DisableDetailedDiff:         pulumi.Bool(false),
			DisableNeoSummaries:         pulumi.Bool(false),
			DisableCodeAccessForReviews: pulumi.Bool(false),
		}); err != nil {
			return err
		}

		if _, err := integrations.NewGitHubEnterpriseIntegration(ctx, "githubEnterprise", &integrations.GitHubEnterpriseIntegrationArgs{
			OrgName:                     pulumi.String(serviceOrg),
			IntegrationId:               pulumi.String(githubEnterpriseID),
			DisablePRComments:           pulumi.Bool(true),
			DisableDetailedDiff:         pulumi.Bool(false),
			DisableNeoSummaries:         pulumi.Bool(false),
			DisableCodeAccessForReviews: pulumi.Bool(true),
		}); err != nil {
			return err
		}

		if _, err := integrations.NewGitLabIntegration(ctx, "gitlab", &integrations.GitLabIntegrationArgs{
			OrgName:             pulumi.String(serviceOrg),
			IntegrationId:       pulumi.String(gitlabID),
			DisablePRComments:   pulumi.Bool(false),
			DisableDetailedDiff: pulumi.Bool(false),
			DisableNeoSummaries: pulumi.Bool(true),
		}); err != nil {
			return err
		}

		if _, err := integrations.NewBitBucketIntegration(ctx, "bitbucket", &integrations.BitBucketIntegrationArgs{
			OrgName:             pulumi.String(serviceOrg),
			IntegrationId:       pulumi.String(bitbucketID),
			DisablePRComments:   pulumi.Bool(false),
			DisableDetailedDiff: pulumi.Bool(false),
			DisableNeoSummaries: pulumi.Bool(false),
		}); err != nil {
			return err
		}

		if _, err := integrations.NewAzureDevOpsIntegration(ctx, "azureDevOps", &integrations.AzureDevOpsIntegrationArgs{
			OrgName:             pulumi.String(serviceOrg),
			IntegrationId:       pulumi.String(azureDevOpsID),
			DisablePRComments:   pulumi.Bool(true),
			DisableDetailedDiff: pulumi.Bool(true),
			DisableNeoSummaries: pulumi.Bool(true),
		}); err != nil {
			return err
		}

		return nil
	})
}

File: examples/v2/vcs-integrations/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
service_org = config.get("serviceOrg") or "service-provider-test-org"
github_id = config.get("githubIntegrationId") or "gh-org-integration"
github_enterprise_id = config.get("githubEnterpriseIntegrationId") or "ghe-org-integration"
gitlab_id = config.get("gitlabIntegrationId") or "gl-org-integration"
bitbucket_id = config.get("bitbucketIntegrationId") or "bb-org-integration"
azure_devops_id = config.get("azureDevOpsIntegrationId") or "ado-org-integration"

ps_v2.integrations.GitHubIntegration(
    "github",
    org_name=service_org,
    integration_id=github_id,
    disable_pr_comments=False,
    disable_detailed_diff=False,
    disable_neo_summaries=False,
    disable_code_access_for_reviews=False,
)

ps_v2.integrations.GitHubEnterpriseIntegration(
    "githubEnterprise",
    org_name=service_org,
    integration_id=github_enterprise_id,
    disable_pr_comments=True,
    disable_detailed_diff=False,
    disable_neo_summaries=False,
    disable_code_access_for_reviews=True,
)

ps_v2.integrations.GitLabIntegration(
    "gitlab",
    org_name=service_org,
    integration_id=gitlab_id,
    disable_pr_comments=False,
    disable_detailed_diff=False,
    disable_neo_summaries=True,
)

ps_v2.integrations.BitBucketIntegration(
    "bitbucket",
    org_name=service_org,
    integration_id=bitbucket_id,
    disable_pr_comments=False,
    disable_detailed_diff=False,
    disable_neo_summaries=False,
)

ps_v2.integrations.AzureDevOpsIntegration(
    "azureDevOps",
    org_name=service_org,
    integration_id=azure_devops_id,
    disable_pr_comments=True,
    disable_detailed_diff=True,
    disable_neo_summaries=True,
)
